front_end/panels/react_devtools/ReactDevToolsView.ts

```diff
@@ -94,38 +91,79 @@ export class ReactDevToolsViewImpl extends UI.View.SimpleView {
       send: (event, payload): void => this.sendMessage(event, payload),
     };
 
-    // To use the custom Wall we've created, we need to also create our own "Bridge" and "Store" objects.
-    this.bridge = ReactDevTools.createBridge(this.wall);
-    this.store = ReactDevTools.createStore(this.bridge);
-
     // Notify backend if Chrome DevTools was closed, marking frontend as disconnected
-    window.addEventListener('beforeunload', () => this.bridge.shutdown());
+    window.addEventListener('beforeunload', () => {
+      console.debug('[ReactDevToolsView] beforeunload');
+      this.bridge?.shutdown()
+    });
 
     SDK.TargetManager.TargetManager.instance().addModelListener(
       ReactDevToolsModel,
-      ReactDevToolsModelEvents.MessageReceived,
-      this.onMessage,
+      ReactDevToolsModelEvents.Initialized,
+      this.onInitialized,
       this,
     );
     SDK.TargetManager.TargetManager.instance().addModelListener(
       ReactDevToolsModel,
-      ReactDevToolsModelEvents.Initialized,
-      this.initialize,
+      ReactDevToolsModelEvents.Destroyed,
+      this.onDestroyed,
       this,
     );
     SDK.TargetManager.TargetManager.instance().addModelListener(
-      SDK.RuntimeModel.RuntimeModel,
-      SDK.RuntimeModel.Events.ExecutionContextDestroyed,
-      this.onExecutionContextDestroyed,
+      ReactDevToolsModel,
+      ReactDevToolsModelEvents.MessageReceived,
+      this.onMessage,
       this,
     );
+
+    this.renderLoader();
```

Comment:
I would expect it to blink for a moment and then disappear, maybe its not worth adding it here?

Reply:
It's not a blink. It takes around half a second, and in some cases might take more so I think it's fine.